lib/suppressions.h

```diff
@@ -66,6 +66,15 @@ class CPPCHECKLIB Suppressions {
             return *this;
         }
 
+        bool operator==(const Suppression& other) const {
+            return errorId == other.errorId &&
+                fileName == other.fileName &&
+                lineNumber == other.lineNumber &&
+                symbolName == other.symbolName &&
+                hash == other.hash &&
+                thisAndNextLine == other.thisAndNextLine;
```

Comment:
hmm not sure what to do here. in theory I like that `operator==` compares all datamembers. It's unexpected if it doesn't. On the other hand I can understand why `matched` is not compared it's not one of the Suppression "parameters".

Reply:
Maybe if we add a method `isSameParameters(const Suppression& other) const` it will be logical that it leaves out `matched` .

Reply:
I had the same dilemma. I like your suggestion of adding a method.